test/gtest/ucp/test_ucp_rma.cc

```diff
@@ -19,290 +23,143 @@ class test_ucp_rma : public test_ucp_memheap {
-                       sizes, 1000, 1);
-    test_message_sizes(static_cast<blocking_send_func_t>(&test_ucp_rma::nonblocking_get_nbi), 
-                       sizes, 1000, 1);
-}
-
-UCS_TEST_P(test_ucp_rma, nbi_med) {
-    size_t sizes[] = { 1000, 3000, 9000, 17300, 31000, 99000, 130000, 0};
-
-    test_message_sizes(static_cast<blocking_send_func_t>(&test_ucp_rma::nonblocking_put_nbi),
-                       sizes, 100, 1);
-    test_message_sizes(static_cast<blocking_send_func_t>(&test_ucp_rma::nonblocking_get_nbi), 
-                       sizes, 100, 1);
-}
-
-UCS_TEST_SKIP_COND_P(test_ucp_rma, nbi_large, RUNNING_ON_VALGRIND) {
-    size_t sizes[] = { 1 * UCS_MBYTE, 3 * UCS_MBYTE, 9 * UCS_MBYTE,
-                       17 * UCS_MBYTE, 32 * UCS_MBYTE, 0};
-
-    test_message_sizes(static_cast<blocking_send_func_t>(&test_ucp_rma::nonblocking_put_nbi),
-                       sizes, 3, 1);
-    test_message_sizes(static_cast<blocking_send_func_t>(&test_ucp_rma::nonblocking_get_nbi), 
-                       sizes, 3, 1);
-}
-
-UCS_TEST_P(test_ucp_rma, nb_small) {
-    size_t sizes[] = { 8, 24, 96, 120, 250, 0};
-
-    test_message_sizes(static_cast<blocking_send_func_t>(&test_ucp_rma::nonblocking_put_nb),
-                       sizes, 1000, 1);
-    test_message_sizes(static_cast<blocking_send_func_t>(&test_ucp_rma::nonblocking_get_nb),
-                       sizes, 1000, 1);
-}
-
-UCS_TEST_P(test_ucp_rma, nb_med) {
-    size_t sizes[] = { 1000, 3000, 9000, 17300, 31000, 99000, 130000, 0};
-
-    test_message_sizes(static_cast<blocking_send_func_t>(&test_ucp_rma::nonblocking_put_nb),
-                       sizes, 100, 1);
-    test_message_sizes(static_cast<blocking_send_func_t>(&test_ucp_rma::nonblocking_get_nb),
-                       sizes, 100, 1);
-}
-
-UCS_TEST_SKIP_COND_P(test_ucp_rma, nb_large, RUNNING_ON_VALGRIND) {
-    size_t sizes[] = { 1 * UCS_MBYTE, 3 * UCS_MBYTE, 9 * UCS_MBYTE,
-                       17 * UCS_MBYTE, 32 * UCS_MBYTE, 0};
-
-    test_message_sizes(static_cast<blocking_send_func_t>(&test_ucp_rma::nonblocking_put_nb),
-                       sizes, 3, 1);
-    test_message_sizes(static_cast<blocking_send_func_t>(&test_ucp_rma::nonblocking_get_nb),
-                       sizes, 3, 1);
-}
+protected:
+    void test_mem_types(send_func_t send_func) {
+        std::vector<std::vector<ucs_memory_type_t> > pairs =
+                ucs::supported_mem_type_pairs();
 
-UCS_TEST_P(test_ucp_rma, nonblocking_put_nbi_flush_worker) {
-    test_blocking_xfer(static_cast<nonblocking_send_func_t>(&test_ucp_rma::nonblocking_put_nbi),
-                       DEFAULT_SIZE, DEFAULT_ITERS,
-                       1, false, false);
-    test_blocking_xfer(static_cast<nonblocking_send_func_t>(&test_ucp_rma::nonblocking_put_nbi),
-                       DEFAULT_SIZE, DEFAULT_ITERS,
-                       1, true, false);
-}
+        for (size_t i = 0; i < pairs.size(); ++i) {
```

Comment:
should it be two-dimensional `for `?

Reply:
i wanted to reuse the function ucs::supported_mem_type_pairs(), and this way code is somewhat shorter